lib/state.go

```diff
@@ -67,8 +68,23 @@ type State struct {
 	// TODO: maybe use https://golang.org/pkg/sync/#Pool ?
 	BPool *bpool.BufferPool
 
-	Vu, Iteration int64
-	Tags          map[string]string
+	Vu                   uint64
+	iteration            int64
+	Tags                 map[string]string
+	ScenarioName         string
+	scenarioVUID         map[string]uint64
+	IncrScIter           func() int64
+	IncrScIterGlobal     func() int64
+	scIterMx             sync.RWMutex
+	scenarioVUIter       map[string]int64
+	scIter, scIterGlobal int64
+}
+
+// Init initializes some private state fields.
+func (s *State) Init() {
+	s.scenarioVUID = make(map[string]uint64)
+	s.scenarioVUIter = make(map[string]int64)
```

Comment:
I see why this is map ... I do think it is needed in the current way things are done, but I hope you can drop it with the proposed refactoring as I truly dislike it :sweat_smile:

Reply:
Why? :smile: I think it makes sense to persist this across VU activations for each scenario, otherwise it would get messy with VU sharing between scenarios.

If this is the same comment as [the previous one](https://github.com/k6io/k6/pull/1863#discussion_r643780977), let's resolve this and discuss it there.